src/CoverCrypt.tsx

```diff
@@ -189,48 +275,141 @@ const CoverCrypt: React.FC<{ kmsToken: string }> = ({ kmsToken }) => {
-              const hr = await decryptDataInKms(row.hr, kmsToken, keysId.masterSecretKeyUID, ACCESS_POLICY);
-              const all = { ...JSON.parse(marketing), ...JSON.parse(hr) };
-              return all;
-            })
-          );
-          setKmsClearData(clear);
-          setKmsEncryptedData(undefined);
-        }
-      }
-    } catch (error) {
-      toastError(error);
+  const handleDecryptLocally = async (): Promise<void> => {
+    if (localEncryptedData && decryptionKey) {
+      const retrievedDecryptionKey = await retrieveDecryptionKey(kmsToken, decryptionKey as string);
+      const clearMarketing: Employee[] = await Promise.all(
+        localEncryptedData.map(async (row) => {
+          try {
+            const marketing = await decryptDataLocally(row.marketing, retrievedDecryptionKey.bytes());
+            const decryptedMarketing = JSON.parse(marketing);
+            return decryptedMarketing;
+          } catch {
+            //
+          }
+        })
+      );
+      const clearHR: Employee[] = await Promise.all(
+        localEncryptedData.map(async (row) => {
+          try {
+            const hr = await decryptDataLocally(row.hr, retrievedDecryptionKey.bytes());
+            const decryptedHR = JSON.parse(hr);
+            return decryptedHR;
+          } catch {
+            //
+          }
+        })
+      );
+
+      const clearEmployee = clearMarketing.map((row, key) => {
+        return {
+          uuid: key,
+          first: row?.first != null ? row.first : "–",
+          last: row?.last != null ? row.last : "–",
+          country: row?.country != null ? row.country : "–",
+          email: clearHR[key]?.email != null ? clearHR[key].email : "–",
+          salary: clearHR[key]?.salary != null ? clearHR[key].salary : "–",
+        };
+      });
+
+      setLocalClearData(clearEmployee as Employee[]);
+      setLocalEncryptedData(undefined);
+    }
+  };
+  const handleDecryptInKms = async (): Promise<void> => {
+    // BROWSER
+    if (localEncryptedData && decryptionKey) {
+      const retrievedDecryptionKey = await retrieveDecryptionKey(kmsToken, decryptionKey as string);
+      const clearMarketing: Employee[] = await Promise.all(
+        localEncryptedData.map(async (row) => {
+          try {
+            const marketing = await decryptDataLocally(row.marketing, retrievedDecryptionKey.bytes());
+            const decryptedMarketing = JSON.parse(marketing);
+            return decryptedMarketing;
+          } catch {
+            //
+          }
+        })
+      );
+      const clearHR: Employee[] = await Promise.all(
+        localEncryptedData.map(async (row) => {
+          try {
+            const hr = await decryptDataLocally(row.hr, retrievedDecryptionKey.bytes());
+            const decryptedHR = JSON.parse(hr);
+            return decryptedHR;
+          } catch {
+            //
+          }
+        })
+      );
```

Comment:
maybe refacto and create only one function to decrypt and get both marketing and HR objects, passing the right decryption function as an argument ?